interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,219 @@
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to pause all other sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="PauseOtherSessions"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to query/modify service properties
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (send)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=org.freedesktop.DBus.Properties
+    member=PropertiesChanged
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`
+
+const mediaHubConnectedPlugAppArmor = `
+# Description: Allow using media-hub service.
+
+#include <abstractions/dbus-session-strict>
+
+dbus (receive, send)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=system
+    path=/
+    interface=org.freedesktop.DBus.ObjectManager
+    peer=(label=unconfined),
+`
+
+const mediaHubPermanentSlotSecComp = `
+recvmsg
+sendmsg
+sendto
+recvfrom
+`
+
+const mediaHubConnectedPlugSecComp = `
+recvmsg
+sendmsg
+sendto
+recvfrom
+`
+
+type MediaHubInterface struct{}
+
+func (iface *MediaHubInterface) Name() string {
+	return "media-hub"
+}
+
+func (iface *MediaHubInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		old := []byte("###SLOT_SECURITY_TAGS###")
+		new := []byte("")
+		if release.OnClassic {
+			// On classic MediaHub will run unconfined
+			new = []byte("unconfined")
+		} else {
+			new = slotAppLabelExpr(slot)
+		}
+		snippet := bytes.Replace([]byte(mediaHubConnectedPlugAppArmor), old, new, -1)
+		return snippet, nil
+	case interfaces.SecuritySecComp:
+		return []byte(mediaHubConnectedPlugSecComp), nil
+	}
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return []byte(mediaHubPermanentSlotAppArmor), nil
+	case interfaces.SecuritySecComp:
+		return []byte(mediaHubPermanentSlotSecComp), nil
+	}
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		old := []byte("###PLUG_SECURITY_TAGS###")
+		new := []byte("")
+		if release.OnClassic {
+			// On classic MediaHub will run unconfined
+			new = []byte("unconfined")
+		} else {
+			new = plugAppLabelExpr(plug)
+		}
```

Comment:
Do you expect that when OnClassic, media-hub will always be the deb or can it be a snap? If only a deb, you can drop the OnClassic stuff because it is already unconfined. When a snap, you actually don't want either/or for the label, you always want the `plugAppLabelExpr(plug)`, but OnClassic you want additional policy for unconfined. However, since only allowing unconfined to talk to media-hub OnClassic doesn't provide a security benefit, please change this function to:
```
func (iface *MediaHubInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
	switch securitySystem {
	case interfaces.SecurityAppArmor:
		old := []byte("###PLUG_SECURITY_TAGS###")
                new = plugAppLabelExpr(plug)
                snippet := bytes.Replace([]byte(mediaHubConnectedSlotAppArmor), old, new, -1)
                return snippet, nil
        }
        return nil, nil
}
```

and change mediaHubPermanentSlotAppArmor to include:
```
# Allow unconfined to talk to us
dbus (receive, send)
    bus=session
    path=/core/ubuntu/media/Service{,/**}
    peer=(label=unconfined),
```